### DISCUSSION TITLE
Node version compatible

### DISCUSSION THREAD

**Discussion by comigit1** [pnp/sp-dev-fx-webparts#2418] [Q&A] [answered]
which node version I have to install in order to build as package?
I am trying to build the following sample
https://github.com/pnp/sp-dev-fx-webparts/tree/main/samples/react-directory

**Comment by DonKirkham** [comment, marked as answer]
The solution uses SPFx 1.11.0, so you can use Node 10.

The SPFx version can be found in the package.json file, as the version for "@microsoft/sp-core-library".  Once you know the SPFx version, you can find which version of NodeJS is needed at the following site: [https://tahoeninjas.blog/2019/12/30/spfx-compatibility-matrix/](https://tahoeninjas.blog/2019/12/30/spfx-compatibility-matrix/) 

Good luck!

**Reply by smchargue** [reply]
I just fundamentally don't understand this development environment, and don't know where to start.  Microsoft's "Getting Started" docs say to install "The latest version of Node.js LTS v14" - which isn't the latest, but we'll go by that.  

react-directory supports SPFx 1.11.0 so that takes you down to node 10,   the react-script-editor part I'm spending the entire day trying to "build" is SPFx 1.10.0, so i guess that takes me down to Node 8?   

How do people work with all these different and competing versions of node, and the 8,000 dependencies they all have.  

It really boggles me completely.  Node dev is the most buck-ass naked emperor i've ever seen in a 30 year development career.  

Really, how can you answer any question without add "Good Luck!" to the end of it.

**Reply by jlemoo** [reply]
Yes, when I'm trying to gulp serve one of the SPFX samples, if there's a problem, it usually has something to do with my version of node. Check out this video https://www.youtube.com/watch?v=TD1rw5kq8H4 where Hugo Bernier explains how to use docker containers with docker desktop to deal with this. I agree, though; this should be easier.

**Reply by phillipharding** [reply]
NVM (node version manager) is what many people use to manage different node environments outside of using docker containers
